Sync/StorageClient.swift

```diff
@@ -580,6 +579,10 @@ open class Sync15StorageClient {
         // In Sync 1.5 it's preferred that we delete the root, not /storage.
         return deleteResource("", f: { $0 })
     }
+    
+    func deleteObject(collection: String, guid: String) -> Deferred<Maybe<StorageResponse<JSON>>> {
+        return deleteResource("/storage/\(collection)/\(guid)", f: { $0 })
```

Comment:
I wonder if the leading slash here might be wrong? eg, https://github.com/mozilla-mobile/firefox-ios/blob/9eea0a004e02ebdce2a6acd4e48b841212aba05b/Sync/StorageClient.swift#L589 doesn't use it. (quickly reading the code though, it *looks* like it doesn't matter, but otherwise I can't understand why you'd get a 404 trying to delete below.

Reply:
👍🏾 I'll take it out and retest just to make sure nothing goes wrong. Just FYI, I added the 404 case because you mentioned that a 404 probably shouldn't be an error. But you might not have had the added context that a get all clients call would happen right before the delete which admittedly changes things. Even after implementing it I'm thinking that, unless there _is_ some race condition or other exceptional event, a 404 should be treated as an error. 😬

Reply:
yeah, it probably does work fine if your testing shows the delete does work. I probably said in the past that if you can't work out if you have migrated, then retry and handling a 404 would be reasonable - but if I 'm reading things correctly, you can be sure you are only doing it once, so probably don't need to handle 404s.